src/ec/packages/ec-component-hero-banner/hero-banner.story.js

```diff
@@ -15,105 +16,97 @@ import heroBanner from './ecl-hero-banner.html.twig';
 import notes from './README.md';
 
 const PrepareBanner = data => {
+  getExtraKnobs(data);
+  if (data.link.link.label) {
+    data.link.link.label = text(
+      'Link label',
+      data.link.link.label,
+      buttonLabels.required
+    );
+  }
+
+  if (data.title) {
+    data.title = text('Title', data.title, buttonLabels.required);
+  }
+
+  if (data.title) {
+    data.title = text('Title', data.title, buttonLabels.required);
+  }
+
+  if (data.description) {
+    data.description = text(
+      'Description',
+      data.description,
+      buttonLabels.required
+    );
+  }
+
+  if (data.centered) {
+    data.centered = boolean('Centered', data.centered, buttonLabels.optianal);
+  }
+
+  if (data.image) {
+    data.image = text('Image', data.image, buttonLabels.optianal);
+  }
   data.link.icon.path = defaultSprite; // eslint-disable-line no-param-reassign
   return data;
 };
+
 storiesOf('Components/Banners/Hero Banner', module)
   .addDecorator(withKnobs)
   .addDecorator(withNotes)
   .addDecorator(withCode)
   .add(
     'image',
-    () =>
-      heroBanner(
-        merge(PrepareBanner(dataImage), {
-          link: {
-            link: {
-              label: text('Link label', dataImage.link.link.label),
-            },
-          },
-          title: text('Title', dataImage.title),
-          description: text('Description', dataImage.description),
-          centered: boolean('Centered', dataImage.centered),
-          image: text('Image', dataImage.image),
-        })
-      ),
+    () => {
+      const bannerImage = PrepareBanner(dataImage);
+
+      return heroBanner(bannerImage);
+    },
     {
       notes: { markdown: notes, json: PrepareBanner(dataImage) },
```

Comment:
MMhh...i have the impression you did not follow the pattern here, you can use here dataImage for how it is, here in the notes, basically

Reply:
Still valid.